Eatery/Views/Eateries/CampusMenuHeaderView.swift

```diff
@@ -0,0 +1,97 @@
+//
+//  CampusMenuHeaderView.swift
+//  Eatery
+//
+//  Created by William Ma on 10/26/19.
+//  Copyright © 2019 CUAppDev. All rights reserved.
+//
+
+import Hero
+import UIKit
+
+protocol CampusMenuHeaderViewDelegate: AnyObject {
+
+    func favoriteButtonPressed(on sender: CampusMenuHeaderView)
+
+}
+
+class CampusMenuHeaderView: UIView {
+
+    private let titleLabel = UILabel()
+    var titleHero: HeroExtension<UILabel> { return titleLabel.hero }
+
+    private let favoriteButton = UIButton()
+    var favoriteHero: HeroExtension<UIButton> { return favoriteButton.hero }
+
+    private let paymentView = PaymentMethodsView()
+    var paymentHero: HeroExtension<PaymentMethodsView> { return paymentView.hero }
+
+    weak var delegate: CampusMenuHeaderViewDelegate?
+
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+
+        titleLabel.isOpaque = false
+        titleLabel.font = .boldSystemFont(ofSize: 34)
+        titleLabel.textColor = .white
+        titleLabel.adjustsFontSizeToFitWidth = true
+        titleLabel.minimumScaleFactor = 0.25
+
+        addSubview(titleLabel)
+        titleLabel.snp.makeConstraints { make in
+            make.leading.bottom.equalToSuperview().inset(16)
+        }
```

Comment:
just to be consistent with our general style, can we move `addSubview` from line 40 to line 39. same for the rest of the views here